handlers.go

```diff
@@ -81,3 +85,27 @@ func checksHandler(c *echo.Context) error {
 
 	return c.JSON(http.StatusOK, checks)
 }
+
+// DELETE /sites/:key
+func deleteSiteHandler(c *echo.Context) error {
+	siteKey := c.P(0)
+	err := removeSite(siteKey)
+
+	if err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, "Error deleting site")
+	}
+
+	return c.NoContent(http.StatusNoContent)
+}
+
+func removeSite(Id string) error {
+	_, err := db.Exec("DELETE FROM sites WHERE id=$1", Id)
+	if err != nil {
+		return err
+	}
+
+	_, err = db.Exec("DELETE FROM checks WHERE siteId=$1", Id)
+
+	delete(AllSites, Id)
```

Comment:
We only want to `delete` here if the above `DELETE FROM checks...` was successful.  If either of the two SQL deletes in this function fail, we dont want to delete from `AllSites` either.